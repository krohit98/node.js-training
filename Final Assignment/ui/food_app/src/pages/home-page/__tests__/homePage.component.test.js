import { render, screen } from '@testing-library/react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HomePage from '../homePage.component';

describe("Unit testing HomePage component",()=>{

    test("Snapshot testing",()=>{
        const renderer = new ShallowRenderer();
        renderer.render(<HomePage />);
        const result = renderer.getRenderOutput();

        expect(result).toMatchSnapshot();
    })

    test("Should render HomePage correctly",()=>{
        render(<HomePage />);
        const heading = screen.getByText("Welcome to Food Service");
        const button = screen.getByText("Discover Food Joints");
        expect(heading).toBeInTheDocument();
        expect(button).toBeInTheDocument();
    })


})

