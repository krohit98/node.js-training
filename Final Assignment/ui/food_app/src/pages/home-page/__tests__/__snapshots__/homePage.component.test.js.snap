// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit testing HomePage component Snapshot testing 1`] = `
<div
  className="pageDiv homePageDiv"
>
  <h1
    className="display-1 pageHeading"
  >
    Welcome to Food Service
  </h1>
  <div
    className="getList-btn-div"
  >
    <button
      className="getList-btn app-btn"
      onClick={[Function]}
    >
      Discover Food Joints
    </button>
  </div>
</div>
`;
